doc/user/unit-testing.md

```diff
@@ -1,132 +1,287 @@
-[doc_client]:https://github.com/operator-framework/operator-sdk/blob/5c50126e7a112d67826894997eca143e12dc165f/doc/user/client.md
-[doc_cr_fake_client]:https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/fake
-[repo_memcached_reconcile]:https://github.com/operator-framework/operator-sdk-samples/blob/4c6934448684a6953ece4d3d9f3f77494b1c125e/memcached-operator/pkg/controller/memcached/memcached_controller.go#L82
-[doc_reconcile]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile#Reconciler
-[code_test_example]: https://github.com/operator-framework/operator-sdk-samples/blob/master/memcached-operator/pkg/controller/memcached/memcached_controller_test.go#L25
+## Testing with 3rd Party Resources
+
+You may have added third-party resources in your operator as described in the [`Advanced Topics section of the user guide`][user_guide]. In order to create a unit-test to test this kind of resources, it is required to add the Schema as it was done for your operator project.
+Following is a snippet code as example using the [`v1.Route`][ocp_doc_v1_route] OpenShift Object Schema.
+
+```go
+
+import (
+	...
+	routev1 "github.com/openshift/api/route/v1"
+	...
+)
+
+// TestMemcachedController runs ReconcileMemcached.Reconcile() against a
+// fake client that tracks a Memcached object.
+func TestMemcachedController(t *testing.T) {	
+    ... 
+    // Register operator types with the runtime scheme.
+    s := scheme.Scheme
+    
+    // Add route Openshift scheme
+    if err := routev1.AddToScheme(s); err != nil {
+        t.Fatalf("Unable to add route scheme: (%v)", err)
+    }
+    
+    // Create the mock for the Route 
+    route := &routev1.Route{
+        ObjectMeta: v1.ObjectMeta{
+            Name:      name,
+            Namespace: namespace,
+            Labels:    getAppLabels(name),
+        },
+    }
+    
+    s.AddKnownTypes(appv1alpha1.SchemeGroupVersion, memcached)
+    
+    // Create a fake client to mock API calls.
+    cl := fake.NewFakeClient(objs...)
+    
+    // Create a ReconcileMemcached object with the scheme and fake client.
+    r := &ReconcileMemcached{client: cl, scheme: s}
+    ...
+}
+```
+
+In this way, you will be able to get the mock object injected into the Reconcile as the following example.
+
+```go
+    route := &routev1.Route{}
+    err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, route)
+    if err != nil {
+        t.Fatalf("get route: (%v)", err)
+    }
+```
+
+## Troubleshooting
+
+### How to increase the verbosity of the logs?
+
+Following is a snippet code as an example to increase the verbosity of the logs in order to better troubleshoot your tests.
+
+```go
+import (
+    ...
+    logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+    ...
+)
+func TestMemcachedController(t *testing.T) {	
+    //dev logs
+    logf.SetLogger(logf.ZapLogger(true))
+    ...
+}
+```
+**NOTE** Recommended just for development purpose.
```

Comment:
Nit: Probably remove this line since unit tests are run while in development.
This won't set the logging verbosity for the actual operator while it's running.
```suggestion
```

Reply:
HI @hasbro17 usually the unit tests are executed by the CI. But I agree that this note is not required. I think it will not a problem anyway too. Tks.